packages/mdc-menu-surface/index.js

```diff
@@ -118,6 +118,16 @@ class MDCMenuSurface extends MDCComponent {
     this.foundation_.setFixedPosition(isFixed);
   }
 
+  /**
+   * Sets the absolute x/y position to position based on. Requires the menu to be hoisted.
+   * @param {number} x
+   * @param {number} y
+   */
+  setAbsolutePosition(x, y) {
+    this.foundation_.setAbsolutePosition(x, y);
+    this.setIsHoisted(true);
```

Comment:
I'm a bit confused by `setHoisted` vs. `hoistMenuToBody`. We're setting `setHoisted` here but not actually enforcing that the menu is hoisted. Also, in the foundation, we seem to blindly assume that if it's hoisted, it's hoisted to the body.

Would it make more sense to _only_ expose a `hoistMenuToBody` method on the component? Or should we be supporting other hoisting scenarios?

Reply:
`hoistMenuToBody` is only on the component as a helper method. The `setHoisted` method is a way to for the foundation to know to calculate the position of the menu based on the body and not relative to an anchoring element. 

I think the current use cases we cover are enough. The menu is either positioned relative to an anchor element, or it has been hoisted to a position that is relative to the body. I think trying to cover the case where the menu is only hoisted to a certain level would be problematic.